### PR TITLE
Oscbc 15 testing datalink by mock

### PR DESCRIPTION
**Pull request by Michail-Antropov**
Add bacnet/datalink unit test and test datalink/cobs

### CODE REVIEW

**Review by skarg — APPROVED** [review]

**Review by gregshue — CHANGES_REQUESTED** [review]
`test/bacnet/datalink/mock/` must use the standard headers instead of local versions.

**Comment by Michail-Antropov** [comment]
> @gregshue This headers from a toolchain. The toolchain may not be installed on the test machine and may not be used for unit testing.

File: test/bacnet/datalink/mock/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	${TST_DIR}/ztest/include
+	./src/
+	)
+
+file(GLOB SRC_TEST ./src/*.c)
+
+add_executable(${PROJECT_NAME}
+	# File(s) under test
+	${SRC_DIR}/bacnet/datalink/datalink.c
+	${SRC_DIR}/../ports/rx62n/bacnet.c
```
Comment (gregshue) [inline test/bacnet/datalink/mock/CMakeLists.txt:41]: No files from `ports/` should be built. They should be mocked instead.

File: test/bacnet/datalink/mock/src/arcnet-mock.c [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+int arcnet_send_pdu(BACNET_ADDRESS *dest, BACNET_NPDU_DATA *npdu_data,
+                    uint8_t *pdu, unsigned pdu_len)
+{
+    ztest_check_expected_value(dest);
+    ztest_check_expected_value(npdu_data);
+    ztest_check_expected_data(pdu, pdu_len);
+    return ztest_get_return_value();
+}
+
+bool check_arcnet_receive_src = true;
```
Comment (gregshue) [inline test/bacnet/datalink/mock/src/arcnet-mock.c:35]: @Michail-Antropov This global symbol should not exist.  If you must configure behavior of a mock then use a function and a header for the mock, and prepend the function with `<subsystem-prefix>_mock_<noun>_<verb>`.

File: src/bacnet/datalink/datalink.c [hunk 3]
```diff
@@ -44,7 +44,7 @@
 #include "bacnet/basic/bbmd6/h_bbmd6.h"
 #include "bacnet/datalink/arcnet.h"
 #include "bacnet/datalink/dlmstp.h"
-#include <string.h>
+#include <strings.h>
```
[the remark this answers is not in the document]
Comment (Michail-Antropov) [reply src/bacnet/datalink/datalink.c:47]: @gregshue  This header need for using strcasecmp() function. strcasecmp() is place in strings.h, not string.h.
Error is happenes when build for native_posix at least.

[the remark this answers is not in the document]
Comment (gregshue) [reply src/bacnet/datalink/datalink.c:47]: @Michail-Antropov Thank you.  Please append line with `/* for strcasecmp() */`

**Review by gregshue — APPROVED** [review]

**Pull request merged**